File: Data Structures & Algorithms/reverse-integer/submission-1.cpp
class Solution {
public:
    int reverse(int x) {
        int MAX = (1<<31)-1;
        int MIN = (-1)*(1<<31);
        std::cout <<"Max:Min=" <<MAX<<":"<<MIN
            << std::endl;

       int result=0;
       while( x != 0 ){
        int lsb = x%10;
        std::cout << lsb <<":";
        x = x/10;
        // check result
        if (result > MAX/10 || \
            (result == MAX/10 && lsb > MAX%10) )
            {
                return 0;
            }
        if (result < MIN/10 || \
            (result == MIN/10 && lsb < MIN%10) )
            {
                return 0;
            }
        result = (result*10) + lsb;
       }

       std::cout << " result " <<std::endl;
       return result;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        table = {}
        for i in nums:
            table[i] = table.get( i, 0) + 1
        # transform frequent elements
        dump = [ (v,k) for k,v in table.items() ]
        dump.sort(reverse=True)
        print(dump)
        # find k most frequent elements
        result = []
        for i in range(k):
            result.append( dump[i][-1] )
        return list(result)
